### ISSUE TITLE
Avoid running e2e and other code relates CI checks when updating docs

### ISSUE THREAD

**Issue by jeanduplessis** [crossplane/crossplane#5242]
### What problem are you facing?

When making a non-code update the full suite of CI checks are run unnecessarily.
See for example this update to the README file: https://github.com/crossplane/crossplane/pull/5231

### How could Crossplane help solve your problem?

Add logic to the CI action to determine whether to run code-related checks.

**Comment by negz** [comment]
https://github.com/crossplane/crossplane/blob/2a648bb1/.github/workflows/ci.yml#L63

Our CI is already supposed to avoid running for changes that don't touch code. Not sure why it ran for #5231. I see the following in the CI log:

```
Stop backtracking for 'global' at commit https://github.com/crossplane/crossplane/commit/f3d49fb646d7fd75627dc926aff707d039fd1133 because '.github/workflows/ci.yml,.github/workflows/scan.yaml,ADOPTERS.md,README.md' are not skippable against paths '' or paths_ignore '**.md,**.png,**.jpg'
```

I'm not sure what that means. My best guess would be that maybe rebasing the PR confused it?

**Comment by jbw976** [comment]
I ran into this about a month ago as well where all of CI ran on a PR of mine that only touched the main README.md (i think).  I looked further into the action we're using to detect no-op and the reason for my case was that when it detects no-op, it will then go back to the most recent commit it can find that had a CI result to examine.  Only if that past result was success will it then consider no-op as a safe thing to do.

This is explained in their algoirthm here: https://github.com/fkirc/skip-duplicate-actions?tab=readme-ov-file#option-3-look-for-successful-checks-of-previous-commits

Not 100% sure this is what bit you too @jeanduplessis, but it's likely.  The more failures we have in the main branch, the more often this can happen 😅
